sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -579,41 +690,83 @@ public Read withTemplateCompatibility() {
         BATCH
     }
 
+    /**
+     * An enumeration type for the method to be used when reading data from BigQuery.
+     */
+    public enum Method {
+
+      /**
+       * The default behavior if no method is explicitly set. Currently {@link #GCS_EXPORT}.
+       */
+      DEFAULT,
+
+      /**
+       * Export data to Google Cloud Storage in Avro format and read data from that location. This
+       * option can be used to read from existing BigQuery tables and to read the results of
+       * queries, and requires that callers specify a temporary Google Cloud Storage bucket to
+       * store the exported data.
+       */
+      GCS_EXPORT,
+
+      /**
+       * Read the contents of a table directly from BigQuery storage using the BigQuery parallel
+       * read API. This option can be used only to read the contents of an existing table.
```

Comment:
I suppose this will eventually get an unbounded version too - will that be a separate method, or a separate configuration property?

Reply:
I haven't thought about the unbounded case at all yet. I'm actually not sure what scenario would require us provide an unbounded source? Standing reads / standing SQL queries / etc. ?

Reply:
Yes, streaming new rows that are added to the table I suppose, if that's an interesting use case. But ok to keep as-is for now.